Parse/Internal/User/CurrentUserController/PFCurrentUserController.m

```diff
@@ -102,9 +103,16 @@ - (BFTask *)_getCurrentUserAsyncWithOptions:(PFCurrentUserLoadingOptions)options
             matchesDisk = _currentUserMatchesDisk;
             currentUser = _currentUser;
         });
-        if (currentUser) {
-            return currentUser;
-        }
+		if (currentUser) {
+			// Verify that the current user matches our expected registered subclass
+			Class expectedClass = [[PFObjectSubclassingController defaultController] subclassForParseClassName:@"_User"];
+			if(expectedClass == [currentUser class]){
```

Comment:
We probably should do `[currentUser isKindOfClass:expectedClass]` here instead of direct class comparison.